### PR TITLE
[ICDS Dashbaord] Fixed the issue where block users are not able to export child beneficiary report

### PR DESCRIPTION
**Pull request by challabeehyv**
https://dimagi-dev.atlassian.net/browse/ICDS-1331

##### SUMMARY
Fixed the issue where block users are not able to export child beneficiary report

##### FEATURE FLAG
No

### CODE REVIEW

**Comment by Rohit25negi** [comment]
This fixes the issue mentioned in the ticket. But I just checked that page locally. It looks location filter on that page is totally messed up which needs to be fixed properly

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 1]
```diff
@@ -246,6 +253,35 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         locationsService.onSelectLocation($item, level, locationsCache, vm);
     };
 
+    vm.onSelect = function($item, level) {
```
Comment (calellowitz) [inline custom/icds_reports/static/js/directives/download/download.directive.js]: I think the place where this is called was meant to be a recursive call in line 245 of the location service, and that should also be fixed there, so other pages using the service have the correct behavior.

Is this just copied back from the old `onSelect`, if so, it should be renamed `onSelectLocation` and can entirely replace the one two lines above

**Review by calellowitz — CHANGES_REQUESTED** [review]
Requested a few small changes and clarifications

**Comment by calellowitz** [comment]
Fixed in https://github.com/dimagi/commcare-hq/pull/27017

**Pull request closed without merging**